collect_app/src/main/java/org/odk/collect/android/fragments/dialogs/MyanmarDatePickerDialog.java

```diff
@@ -0,0 +1,77 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.odk.collect.android.fragments.dialogs;
+
+import org.javarosa.core.model.FormIndex;
+import org.joda.time.LocalDateTime;
+import org.odk.collect.android.logic.DatePickerDetails;
+import org.odk.collect.android.utilities.MyanmarDateUtils;
+
+import mmcalendar.MyanmarDate;
+import mmcalendar.MyanmarDateKernel;
+
+public class MyanmarDatePickerDialog extends CustomDatePickerDialog {
+    private static final int MIN_SUPPORTED_YEAR = 1261; //1900 in Gregorian calendar
+    private static final int MAX_SUPPORTED_YEAR = 1462; //2100 in Gregorian calendar
+
+    public static MyanmarDatePickerDialog newInstance(FormIndex formIndex, LocalDateTime date, DatePickerDetails datePickerDetails) {
+        MyanmarDatePickerDialog dialog = new MyanmarDatePickerDialog();
+        dialog.setArguments(getArgs(formIndex, date, datePickerDetails));
+
+        return dialog;
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        setUpDatePicker();
+        updateGregorianDateLabel();
+    }
+
+    @Override
+    protected void updateDays() {
+        MyanmarDate myanmarDate = getCurrentMyanmarDate();
+        setUpDayPicker(MyanmarDateUtils.getFirstDayOfCurrentMonth(myanmarDate), myanmarDate.getMonthDay(), myanmarDate.getMonthLength());
+    }
+
+    @Override
+    protected void yearUpdated() {
+        MyanmarDate myanmarDate = getCurrentMyanmarDate();
+        setUpMonthPicker(MyanmarDateUtils.getMonthId(myanmarDate) + 1, MyanmarDateUtils.getMyanmarMonthsArray(myanmarDate.getYearInt()));
+        super.yearUpdated();
+    }
+
+    @Override
+    protected LocalDateTime getOriginalDate() {
+        return MyanmarDateUtils.myanmarDateToGregorianDate(getCurrentMyanmarDate());
+    }
+
+    private void setUpDatePicker() {
+        MyanmarDate myanmarDate = MyanmarDateUtils.gregorianDateToMyanmarDate(getDateWithSkippedDaylightSavingGapIfExists());
+
+        setUpDayPicker(MyanmarDateUtils.getFirstDayOfCurrentMonth(myanmarDate), myanmarDate.getMonthDay(), myanmarDate.getMonthLength());
+        setUpMonthPicker(MyanmarDateUtils.getMonthId(myanmarDate) + 1, MyanmarDateUtils.getMyanmarMonthsArray(myanmarDate.getYearInt()));
+        setUpYearPicker(myanmarDate.getYearInt(), MIN_SUPPORTED_YEAR, MAX_SUPPORTED_YEAR);
+    }
+
+    private MyanmarDate getCurrentMyanmarDate() {
+        return MyanmarDateUtils.createMyanmarDate(
+                getYear(),
+                MyanmarDateKernel.getMyanmarMonth(getYear(), 1).getIndex().get(getMonthId()),
```

Comment:
I encountered a crash while scrolling up and down in the date picker with `myanmar.xml`. I think it happened when changing the year after having already changed the month and day.

```console
    java.lang.IndexOutOfBoundsException: Index: 13, Size: 13
        at java.util.ArrayList.get(ArrayList.java:437)
        at org.odk.collect.android.fragments.dialogs.MyanmarDatePickerDialog.getCurrentMyanmarDate(MyanmarDatePickerDialog.java:74)
        at org.odk.collect.android.fragments.dialogs.MyanmarDatePickerDialog.updateDays(MyanmarDatePickerDialog.java:47)
        at org.odk.collect.android.fragments.dialogs.CustomDatePickerDialog.yearUpdated(CustomDatePickerDialog.java:224)
        at org.odk.collect.android.fragments.dialogs.MyanmarDatePickerDialog.yearUpdated(MyanmarDatePickerDialog.java:55)
        at org.odk.collect.android.fragments.dialogs.CustomDatePickerDialog$5.onValueChange(CustomDatePickerDialog.java:149)
```

Unable to repro after the first time, it's probably for a very specific combination of days.

Reply:
Thank, I reproduced and fixed the bug.